# Python Loops

In [2]:
# Why Loops Matter

# Instead of writing:
print(1)
print(2)
print(3)
print(4)

# We automate repetition.

1
2
3
4


In [7]:
# Topic 1: for Loop
for i in range(5):
    print(i)

0
1
2
3
4


In [ ]:
range(5)
# 5 is excluded

range(0, 5)

In [8]:
# Topic 2: Controlled Range
for i in range (1,5):
    print(i)

1
2
3
4


In [9]:
# Topic 3: Step Size
for i in range(0,11,2):
    print(i)

0
2
4
6
8
10


In [ ]:
# Topic 4: Using Loop Variables
for t in range(5):
    temperature = 300 + 10*t
    print(temperature)

# This mimics heating over time steps.

300
310
320
330
340


In [12]:
# Topic 5: while Loop

# Use when repetition depends on condition

x = 0

# while x < 5:
#     print(x) # No update → never ends.. this is infinite cell, stop the running cell. 
while x < 5:
    x = x + 1
    print(x)

1
2
3
4
5


# Exercises

Write programs:

Exercise 1

Print numbers 1 to 10

Exercise 2

Print squares of 1 to 5

Exercise 3

Print temperatures:

300, 320, 340, 360, 380

Exercise 4

Use while loop to print 5 to 1